src/Processors/Formats/Impl/TabSeparatedRowInputFormat.cpp

```diff
@@ -89,30 +95,34 @@ void TabSeparatedFormatReader::skipFieldDelimiter()
 
 void TabSeparatedFormatReader::skipRowEndDelimiter()
 {
+    bool supports_crlf = format_settings.tsv.crlf_end_of_line_input;
     if (buf->eof())
         return;
-
+    if (supports_crlf && first_row==false)
+    {
+        ++buf->position();
+    }
     if (unlikely(first_row))
     {
-        checkForCarriageReturn(*buf);
+        supports_crlf ? checkForCarriageReturn<true>(*buf) : checkForCarriageReturn<false>(*buf);
         first_row = false;
     }
-
     assertChar('\n', *buf);
 }
 
 template <bool read_string>
 String TabSeparatedFormatReader::readFieldIntoString()
 {
     String field;
+    bool support_crlf = format_settings.tsv.crlf_end_of_line_input;
     if (is_raw)
         readString(field, *buf);
     else
     {
         if constexpr (read_string)
-            readEscapedString(field, *buf);
+            support_crlf ? readEscapedStringCRLF<true>(field, *buf) : readEscapedStringCRLF<false>(field, *buf);
```

Comment:
```suggestion
            support_crlf ? readEscapedStringCRLF(field, *buf) : readEscapedString(field, *buf);
```
Let's make `readEscapedStringCRLF` not template and use this